adapters/repos/db/shard_dimension_tracking.go

```diff
@@ -126,6 +126,10 @@ func (s *Shard) initDimensionTracking() {
 		// leak. The actual work should be much faster.
 		ctx, cancel := context.WithTimeout(rootCtx, 30*time.Minute)
 		defer cancel()
+		defer func() {
+			s.dimensionTrackingInitialized.Store(true)
```

Comment:
might just be better to be at the beginning (without deferring it), otherwise if there is a quit shutdown before ending this function (rare but still) could do a leak of publishing routine

Reply:
Good point, I'll move it.